### PR TITLE
add validators for command line arguments that require paths

### PR DESCRIPTION
**Pull request by ftyers**
This adds a check to make sure files exist using abseil for some of the command line flags. See issue in  #1954.

### CODE REVIEW

File: util/flags.py [hunk 1]
```diff
@@ -113,3 +113,21 @@ def create_flags():
...
+    f.register_validator('lm_trie_path',
+                         lambda value: os.path.isfile(value),
+                         message='The file pointed to by --lm_trie_path must exist and be readable.')
+
+    f.register_validator('alphabet_config_path',
+                         lambda value: os.path.isfile(value),
+                         message='The file pointed to by --alphabet_config_path must exist and be readable.')
+
+    f.register_validator('one_shot_infer',
+                         lambda value: os.path.isfile(value),
```
Comment (reuben) [inline util/flags.py]: Seems like validators are always run, even for default values. So for optional flags you need to check that case, or it errors out whenever you don't specify `--one_shot_infer`. You can change the checker to be:

```python
lambda value: not value or os.path.isfile(value)
```

File: util/flags.py [hunk 2]
```diff
@@ -113,3 +113,21 @@ def create_flags():
     # Inference mode
 
     f.DEFINE_string('one_shot_infer', '', 'one-shot inference mode: specify a wav file and the script will load the checkpoint and perform inference on it.')
+
+    # Register validators for paths which require a file to be specified
+
+    f.register_validator('lm_binary_path',
+                         lambda value: os.path.isfile(value),
```
[the remark this answers is not in the document]
Comment (reuben) [reply util/flags.py]: Looks like Pylint also caught this :)

[the remark this answers is not in the document]
Comment (ftyers) [reply util/flags.py]: Oops, I forgot to run pylint, I'll do it in future!

**Comment by ftyers** [comment]
I've taken care of those two changes. Thanks for the invite! What is the right thing to do with your comments now ? "Resolve conversation" ?

**Comment by ftyers** [comment]
I also added a check for `--export_dir`, but accidentally added whitespace after removing it and squashing the commit I messed up the commit message, sorry about that!

**Comment by reuben** [comment]
Yeah clicking resolve conversation is fine, but it's more for your own tracking of what was fixed.

**Comment by reuben** [comment]
Don't worry about the commit message, I'll just use squash and merge on the GitHub UI when the tests have run.

**Pull request merged**